java/integTest/src/test/java/glide/PubSubTests.java

```diff
@@ -570,19 +604,96 @@ public void combined_exact_pattern_and_sharded_one_client(boolean useCallback) {
 
         messages.addAll(shardedMessages);
         verifyReceivedPubsubMessages(
-                messages.stream().map(m -> Pair.of(1, m)).collect(Collectors.toSet()),
-                listener,
-                useCallback);
+                messages.stream().map(m -> Pair.of(1, m)).collect(Collectors.toSet()), listener, method);
+    }
+
+    @SneakyThrows
+    @Test
+    public void coexistense_of_sync_and_async_read() {
```

Comment:
This test fully covers
* `test_pubsub_pattern_co_existence`
* `test_sharded_pubsub_co_existence`
* `test_pubsub_exact_happy_path_many_channels_co_existence` which covers `test_pubsub_exact_happy_path_coexistence`